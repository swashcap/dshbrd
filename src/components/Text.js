// @flow
import * as React from "react";
import styled from "styled-components";

type Props = {
  children?: React.Node,
  tag: string,
  size: "body" | "caption" | "heading" | "subheading"
};

const TextBase = styled(({ children, tag, ...props }) =>
  React.createElement(tag, props, children)
)`
  font-family: "Avenir Next", "Helvetica Neue", Arial, sans-serif;
  font-size: 100%;
  line-height: 1.5;
`;

const TextBody = TextBase;

const TextCaption = styled(TextBase)`
  font-size: 75%;
  line-height: 1.25;
`;

const TextHeading = styled(TextBase)`
  font-size: 150%;
  font-weight: 700;
  line-height: 1.25;
`;

const TextSubheading = styled(TextBase)`
  font-size: 112.5%;
  font-weight: 700;
  line-height: 1.333;
`;

export default class Text extends React.Component<Props> {
  static defaultProps = {
    size: "body",
    tag: "div"
  };

  render() {
    const { children, size, tag, ...rest } = this.props;

    if (size === "heading") {
      return (
        <TextHeading tag={tag} {...rest}>
          {children}
        </TextHeading>
      );
    } else if (size === "subheading") {
      return (
        <TextSubheading tag={tag} {...rest}>
          {children}
        </TextSubheading>
      );
    } else if (size === "caption") {
      return (
        <TextCaption tag={tag} {...rest}>
          {children}
        </TextCaption>
      );
    }

    return (
      <TextBody tag={tag} {...rest}>
        {children}
      </TextBody>
    );
  }
}
