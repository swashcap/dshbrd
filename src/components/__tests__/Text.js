// @flow
import * as React from "react";
import { mount, shallow } from "enzyme";
import TestRenderer from "react-test-renderer";

import Text from "../Text";

test("renders correctly", () => {
  const testRenderer = TestRenderer.create(<Text>Sample text</Text>);
  expect(testRenderer).toMatchSnapshot();
});

test("renders div by default", () => {
  const wrapper = mount(<Text>Hi</Text>);
  const div = wrapper.find("div");

  expect(div.exists()).toBe(true);
  expect(div.text()).toBe("Hi");
});

test("renders custom tag", () => {
  const wrapper = mount(<Text tag="h1">Sample text</Text>);
  expect(wrapper.find("div").exists()).toBe(false);
  expect(wrapper.find("h1").text()).toBe("Sample text");
});

// test("it renders", () => {
//   const testRenderer = TestRenderer.create(<Text>Sample text</Text>);
//
//   expect(testRenderer).toMatchSnapshot();
// });

// test("passes tag", () => {
//   const renderer = new ShallowRenderer();
//   renderer.render(<Text>Sample text</Text>);
//   const result = renderer.getRenderOutput();
//   debugger;
//
//   expect(result.type).toBe("div");
// });
