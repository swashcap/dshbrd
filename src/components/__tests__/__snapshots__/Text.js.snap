// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  font-family: "Avenir Next","Helvetica Neue",Arial,sans-serif;
  font-size: 100%;
  line-height: 1.5;
}

<div
  className="c0"
>
  Sample text
</div>
`;
